keps/762-ResourceClaimTemplate/README.md

```diff
@@ -0,0 +1,417 @@
+* Allow optional targeting of specific containers within the Pod to receive access to the shared ResourceClaim.
+* Manage the lifecycle of the Job-scoped ResourceClaims, ensuring they are
+created when the Job starts and deleted when the Job completes or is deleted.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Modifying the core Kubernetes Job/ PodSpec API.
+* Supporting retroactive application of new or modified `resourceClaimTemplates` to already running Jobs.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Introduce a new field `resourceClaimTemplates` within the `JobSet.spec`. This
+field will be a list of `ResourceClaimTemplate` objects, similar to how they are defined
+within a Pod's spec, but at the JobSet level.
+
+
+### Design Details
+
+#### API Changes
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+```yaml
+
+# Example Pod-level ResourceClaimTemplate to contrasnt with per Job
+# ResourceCLaimTemplate below.
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaimTemplate
+metadata:
+  name: gpu
+spec:
+  devices:
+    requests:
+    - name: gpu
+      deviceClassName: gpu.nvidia.com
+---
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: shared-resource-jobset
+spec:
+  replicatedJobs:
+    - name: worker-group
+      replicas: 3
+      template: # This is the JobTemplateSpec
+        spec: # This is the JobSpec
+          parallelism: 1 # JobSpec fields
+          completions: 1 # JobSpec fields
+          template: # This is the PodTemplateSpec
+            spec: # This is the PodSpec
+              containers:
+                - name: worker
+                  image: your-training-image:latest
+                  command: ["python", "train.py"]
+                  resources:
+                    requests:
+                      cpu: "2"
+                      memory: "4Gi"
+                  resourceClaims: # Pod-level DRA
+                    - name: gpu-claim
+                      source:
+                        resourceClaimTemplateName: pod-gpu-template
+                - name: helper
+                  image: helper-image:latest
+                  command: ["sleep", "infinity"]
+              restartPolicy: OnFailure
+          resourceClaimTemplates:  # NEW: JobSet-level ResourceClaimTemplates (added inside JobSpec)
```

Comment:
Why this API is under `replicatedJob[].template.spec.resourceClaimTemplates` ?

Reply:
Ah good catch! This should be one level higher 😛